test/bdd/agent_controller_steps.go

```diff
@@ -89,6 +94,49 @@ func (a *AgentWithControllerSteps) checkAgentIsRunning(agentID, inboundHost, inb
 	logger.Infof("Webhook for agent '%s' is running on '%s''", agentID, webhookURL)
 	a.webhookURLs[agentID] = webhookURL
 
+	go a.pullWebhookEvents(agentID)
+
+	return nil
+}
+
+func (a *AgentWithControllerSteps) registerForStateNotification(agentID, state string, ch chan string) error {
+	a.stateNotificationMu.Lock()
+
+	if a.stateNotificationCh[agentID] == nil {
+		a.stateNotificationCh[agentID] = make(map[string]chan string)
+	}
+	a.stateNotificationCh[agentID][state] = ch
+
+	a.stateNotificationMu.Unlock()
+
+	return nil
+}
+
+func (a *AgentWithControllerSteps) pullWebhookEvents(agentID string) error {
+	webhookURL, ok := a.webhookURLs[agentID]
+	if !ok {
+		return fmt.Errorf("unable to find webhook URL for agent [%s]", webhookURL)
+	}
+
+	// try to pull recently pushed topics from webhook
+	var connectionMsg didexchrsapi.ConnectionMsg
+	for i := 0; i < pullTopicsAttempts; i++ {
+		err := sendHTTP(http.MethodGet, webhookURL+checkForTopics, nil, &connectionMsg)
```

Comment:
I see this is just a refactor from another PR, but why do we need to pull from a WebHook URL instead of listening to them?

@rolsonquadras @sudeshrshetty @DRK3

Reply:
1. No way to get connection ID now since receive-invitation is returning fake connection ID.
2. Way to verify REST API sending notifications to webhook

Reply:
What is the issue number for fixing the "fake connection ID"? I assume we are fixing it soon.

Reply:
#537 Already in TODO of restapi/didexchange

Reply:
https://github.com/hyperledger/aries-framework-go/blob/master/pkg/restapi/operation/didexchange/didexchange.go#L147

Reply:
@troyronda Current bdd structure has 3 components- agent, webhook server and business logic(bdd-test). 

1. agent sends callback to webhook-server.
2. bdd-test polls webhook-server and retrieves the callback messages from webhook-server.
3. bdd-test calls agent for approving the request.

The question is whether we need to have a separate webhook-server ? @sudeshrshetty

Reply:
@rolsonquadras Yes, no other way to verify if webhooks servers are getting notifications from  controller API logic.

Reply:
@sudeshrshetty Can't we have webhook server as part of bdd-test and integrate with business logic rather than going through extra hop ?